packages/react-dom/src/__tests__/ReactDOMComponentTree-test.js

```diff
@@ -10,32 +10,42 @@
 'use strict';
 
 describe('ReactDOMComponentTree', () => {
-  var React;
-  var ReactDOM;
-  var ReactDOMComponentTree;
-  var ReactDOMServer;
+  let React;
+  let ReactDOM;
+  let ReactDOMServer;
 
   function renderMarkupIntoDocument(elt) {
-    var container = document.createElement('div');
+    const container = document.createElement('div');
     // Force server-rendering path:
     container.innerHTML = ReactDOMServer.renderToString(elt);
     return ReactDOM.hydrate(elt, container);
   }
 
-  function getTypeOf(instance) {
-    return instance.type;
+  function simulateInput(elem, value) {
+    const inputEvent = new Event('input', {
+      bubbles: true,
+    });
+    setUntrackedInputValue.call(elem, value);
```

Comment:
Can you remind me why this is necessary?

Reply:
In order to simulate the behavior of trying to switch a component from being uncontrolled to controlled we need to assign a new value to the input without it being tracked - I found this [approach](https://github.com/facebook/react/pull/11309#pullrequestreview-70996771) after reading through the comments of #11309.